pkg/controller/controller.go

```diff
@@ -111,14 +111,13 @@ func (c *Controller) Run(ctx context.Context) {
 // getKubeClient returns a kubeclient depending if the ccm runs inside a container
 // inside the same docker network that the kind cluster or run externally in the host
 // It tries first to connect to the external endpoint
-func (c *Controller) getKubeClient(ctx context.Context, cluster string) (kubernetes.Interface, error) {
+func (c *Controller) getKubeClient(ctx context.Context, cluster string) (kubernetes.Interface, bool, error) {
```

Comment:
this should maybe be an "enum", there's host, in-network IP, in-network container name/DNS, etc ...

it also feels like a bit of a conflation of concerns to return a kubernetes client AND the network connectivity mode here